src/toil/jobStores/abstractJobStore.py

```diff
@@ -132,6 +132,39 @@ class AbstractJobStore(object):
     """
     __metaclass__ = ABCMeta
 
+    class Locator(object):
+        @classmethod
+        def parse(cls, s):
+            l = s.split(':')
```

Comment:
Not what we talked about. You can't assume that subclasses separate their locator components with ':'. You only know about one particular colon. The first one.